src/ripple/server/impl/ServerHandlerImp.cpp

```diff
@@ -252,29 +251,37 @@ ServerHandlerImp::onWSMessage(
     std::shared_ptr<WSSession> session,
         std::vector<boost::asio::const_buffer> const& buffers)
 {
-    // VFALCO This is inefficient, the JSON
-    // should be parsed from the buffer sequence.
-    std::string s;
-    s.reserve(boost::asio::buffer_size(buffers));
-    std::copy(boost::asio::buffers_begin(buffers),
-        boost::asio::buffers_end(buffers),
-            std::back_inserter(s));
-//m_journal.error << "Recv: " << s;
     Json::Value jv;
-    // VFALCO should we parse a coroutine instead?
-    if(! Json::Reader{}.parse(s, jv))
-    {
-        // TODO Send error
+    auto const size = boost::asio::buffer_size(buffers);
+    if (size > RPC::Tuning::maxRequestSize ||
+        ! Json::Reader{}.parse(jv, buffers) ||
```

Comment:
Would it make sense to make `parse` a static member of `Json::Reader`? Calling the member like this is a little awkward.